src/components/node-list/node-list.scss

```diff
@@ -101,11 +101,89 @@
   margin: 0 12px;
   padding: 0;
   list-style: none;
+
+  .kui-theme--light & {
+    background: $color-bg-light-alt2;
+  }
+
+  .kui-theme--dark & {
+    background: $color-bg-dark-alt2;
+  }
+
+  &--nested {
+    margin: 0 0 12px;
+  }
+
+  &--fade-in {
+    animation: fadeIn 0.4s forwards;
+  }
+
+  &--fade-out {
+    animation: fadeIn 0.4s backwards reverse;
+  }
+}
+
+@keyframes fadeIn {
+  0% {
+    opacity: 0;
+  }
+  100% {
+    opacity: 1;
+  }
+}
+
+.pipeline-type-group-toggle {
+  $size: 30px;
+  position: absolute;
+  right: 5px;
+  top: 50%;
+  width: $size;
+  height: $size;
+  line-height: 1em;
+  text-align: center;
+  z-index: 1;
+  padding: 0;
+  margin: -$size/2 0 0;
+  border-radius: 50%;
+  font-size: 1.5em;
+  background: none;
+  border: none;
+  box-shadow: none;
+  color: inherit;
+  font-family: inherit;
+  cursor: pointer;
+  transition: transform ease 0.3s;
```

Comment:
Can this be structured a bit more e.g. outside in ordering

```
.declaration-order {
  /* Positioning */
  position: absolute;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  z-index: 10;
 
  /* Box Model */
  display: block;
  float: right;
  width: 100px;
  height: 100px;
  margin: 10px;
  padding: 10px;
 
  /* Typography */
  color: #888;
  font: normal 16px Helvetica, sans-serif;
  line-height: 1.3;
  text-align: center;
 
  /* Visual */
  background-color: #eee;
  border: 1px solid #888;
  border-radius: 4px;
  opacity: 1;
 
  /* Animation */
  transition: all 1s;
 
  /* Misc */
  user-select: none;
}
```

This can do auto done by stylelint. Would be great to structure these consistently so it's not a big chunk of styling (harder to approach when changing in future)

Reply:
🤷‍♂  I've tried to order these in a way that made sense to me, more or less. e.g. I put positioning at the top, then display, etc. I think it's broadly fine as is? Happy to introduce stylelint to do it automatically, but that can be a separate PR. In the meantime I don't think it's worth doing manually.

Reply:
I'll create an issue to do it automatically, we should have consistency across the style ordering and structure 👍

Reply:
Issue here - https://github.com/quantumblacklabs/kedro-viz/issues/102